### DISCUSSION TITLE
Regularization types

### DISCUSSION THREAD

**Discussion by ghwilliams** [gimli-org/pyGIMLi#580] [Q&A]
Guys I'm having a hard time trying to find the options/descriptions for the regularization types avaiable in PyGIMLi. Maybe the documentation need to be improved in this particular point. By the way, although PyGIMLi looks like a great project, the documentation is a bit opaque in general. 

Thanks
Williams

**Comment by halbmy** [comment]
There is a specific tutorial on regularization https://www.pygimli.org/_tutorials_auto/3_inversion/plot_5_Regularization.html

**Comment by halbmy** [comment]
Plus one on region-wise regularization: https://www.pygimli.org/_tutorials_auto/3_inversion/plot_8-regionWise.html

**Reply by ghwilliams** [reply]
Thank you very much.
These tutorials are interesting but they don't make clear if there are any options other than the ones used in the tutorials. Regarding the options for the regularization, they are a bit like a black box.
Is there a place where the descriptions for all the avaiable options (e.g. for ctype) are described?

**Reply by makeabhishek** [reply]
Hi @ghwilliams,

You may find this paper useful to understand how they set up different regularization terms in the loss (eq. 6). 

**paper:** [Wagner, F. M., Mollaret, C., Günther, T., Kemna, A., & Hauck, C. (2019). Quantitative imaging of water, ice and air in permafrost systems through petrophysical joint inversion of seismic refraction and electrical resistivity data. Geophysical Journal International, 219(3), 1866-1875.](https://academic.oup.com/gji/article/219/3/1866/5559534)

**code:** https://github.com/florian-wagner/four-phase-inversion/blob/master/code/fpinv/jointmod.py#L89-L145

https://github.com/florian-wagner/four-phase-inversion/blob/master/code/fpinv/jointinv.py#L46-L54

Furthermore, in `Pygimli`, the smoothness matrix is a sparse matrix defining `1st` or `2nd` (or `0th` or `mixed` of them) order derivatives.

In the context of pygimli, Tikhonov regularisation. is defined by $C m$, where $C$ is smoothness matrix. We need to define ‘cType’ which is smoothness of different type for example `cType =1` is a first order. So this should be Tikonov regularization. 

* please correct me if there is any mistake.

**Comment by halbmy** [comment]
There is a tutorial about regularization:
https://www.pygimli.org/_tutorials_auto/3_inversion/plot_5_Regularization.html
